ui/app/src/components/shared/ResourceCard.tsx

```diff
@@ -40,122 +40,116 @@ export const ResourceCard: React.FunctionComponent<ResourceCardProps> = (props:
       || (props.resource.azureStatus?.powerState && props.resource.azureStatus.powerState !== VMPowerStates.Running)
   }
 
-  const resourceCosts = costsCtx?.costs?.find((resourceCost) => {
-    return resourceCost.id === props.resource.id;
-  });
-
-  let resourceUrl = ""
-  switch(props.resource.resourceType) {
-    case ResourceType.Workspace:
-    case ResourceType.WorkspaceService:
-    case ResourceType.UserResource:
-      resourceUrl = props.resource.resourcePath;
-      break;
-    case ResourceType.SharedService: // shared services are accessed from the root and the workspace, have to handle the URL differently
-      resourceUrl = workspaceCtx.workspace ? props.resource.id : props.resource.resourcePath;
-      break;
+  const goToResource = useCallback(() => {
```

Comment:
`nit` the `use` hooks want to go above any imperative code - think should move to line 34

Reply:
Fixed